servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncCompletableOperator.java

```diff
@@ -292,27 +321,33 @@ public void onError(final Throwable t) {
      */
     static final class OffloadedCancellable implements Cancellable {
         private final Cancellable cancellable;
+        private final BooleanSupplier offload;
         private final Executor executor;
 
-        OffloadedCancellable(final Cancellable cancellable, final Executor executor) {
+        OffloadedCancellable(final Cancellable cancellable, final BooleanSupplier offload, final Executor executor) {
             this.cancellable = requireNonNull(cancellable);
+            this.offload = offload;
             this.executor = executor;
         }
 
         @Override
         public void cancel() {
-            try {
-                LOGGER.debug("executing Completable cancel on {}", executor);
-                executor.execute(() -> safeCancel(cancellable));
-            } catch (Throwable t) {
-                LOGGER.warn("Failed to execute task on the executor {}. " +
-                                "Invoking Cancellable (cancel()) in the caller thread. Cancellable {}. ",
-                        executor, cancellable, t);
-                // As a policy, we call the target in the calling thread when the executor is inadequately
-                // provisioned. In the future we could make this configurable.
+            if (offload.getAsBoolean()) {
+                try {
+                    LOGGER.debug("executing Completable cancel on {}", executor);
```

Comment:
can we remove the debug/development logs? we can always instrument with operators if necessary (e.g. `afterCancel(...)`).